cmd/sks_nodepool_show.go

```diff
@@ -96,100 +100,100 @@ func (c *sksNodepoolShowCmd) cmdRun(_ *cobra.Command, _ []string) error {
 
 	out := sksNodepoolShowOutput{
 		AddOns: func() (v []string) {
-			if nodepool.AddOns != nil {
-				v = *nodepool.AddOns
+			if nodepool.Addons != nil {
+				v = nodepool.Addons
 			}
 			return
 		}(),
 		AntiAffinityGroups: make([]string, 0),
-		CreationDate:       nodepool.CreatedAt.String(),
-		Description:        utils.DefaultString(nodepool.Description, ""),
-		DiskSize:           *nodepool.DiskSize,
-		ID:                 *nodepool.ID,
-		InstancePoolID:     *nodepool.InstancePoolID,
-		InstancePrefix:     utils.DefaultString(nodepool.InstancePrefix, ""),
+		CreationDate:       nodepool.CreatedAT.String(),
+		Description:        utils.DefaultString(&nodepool.Description, ""),
+		DiskSize:           nodepool.DiskSize,
+		ID:                 nodepool.ID,
+		InstancePoolID:     nodepool.InstancePool.ID,
+		InstancePrefix:     utils.DefaultString(&nodepool.InstancePrefix, ""),
 		Labels: func() (v map[string]string) {
 			if nodepool.Labels != nil {
-				v = *nodepool.Labels
+				v = nodepool.Labels
 			}
 			return
 		}(),
-		Name:            *nodepool.Name,
+		Name:            nodepool.Name,
 		SecurityGroups:  make([]string, 0),
 		PrivateNetworks: make([]string, 0),
-		Size:            *nodepool.Size,
-		State:           *nodepool.State,
+		Size:            nodepool.Size,
+		State:           string(nodepool.State),
 		Taints: func() (v []string) {
 			if nodepool.Taints != nil {
 				v = make([]string, 0)
-				for k, t := range *nodepool.Taints {
+				for k, t := range nodepool.Taints {
 					v = append(v, fmt.Sprintf("%s=%s:%s", k, t.Value, t.Effect))
 				}
 			}
 			return
 		}(),
 		ImageGCMin: func() (s string) {
-			if nodepool.KubeletImageGc != nil && nodepool.KubeletImageGc.MinAge != nil {
-				return *nodepool.KubeletImageGc.MinAge
+			if nodepool.KubeletImageGC != nil && nodepool.KubeletImageGC.MinAge != "" {
+				return nodepool.KubeletImageGC.MinAge
 			}
 			return kubeletImageGcMinAge
 		}(),
 		ImageGcLowThreshold: func() (s int64) {
-			if nodepool.KubeletImageGc != nil && nodepool.KubeletImageGc.LowThreshold != nil {
-				return *nodepool.KubeletImageGc.LowThreshold
+			if nodepool.KubeletImageGC != nil && nodepool.KubeletImageGC.LowThreshold != 0 {
```

Comment:
not sure, that checking for 0 is the right way..

Reply:
Zero is uninitialized value of int64 so your code is correct.